web-api/setup-migration-user.sh

```diff
@@ -0,0 +1,149 @@
+  "section": "$section",
+  "name": "$name",
+  "firstName": "$firstName",
+  "middleName": "$middleName",
+  "lastName": "$lastName",
+  "suffix": "$suffix",
+  "barNumber": "$barNumber",
+  "admissionsDate": "2019-03-01T21:40:46.415Z",
+  "admissionsStatus": "Active",
+  "birthYear": "1950",
+  "employer": "$employer",
+  "firmName": "Some Firm",
+  "originalBarState": "WA",
+  "practitionerType": "Attorney",
+  "contact": {
+    "address1": "234 Main St",
+    "address2": "Apartment 4",
+    "address3": "Under the stairs",
+    "city": "Chicago",
+    "countryType": "domestic",
+    "phone": "+1 (555) 555-5555",
+    "postalCode": "61234",
+    "state": "IL"
+  }
+}
+EOF
+}
+
+createAdmin() {
+  email=$1
+  role=$2
+  name=$3
+
+  aws cognito-idp sign-up \
+    --region "${REGION}" \
+    --client-id "${CLIENT_ID}" \
+    --username "${email}" \
+    --user-attributes 'Name="name",'Value="${name}" 'Name="custom:role",'Value="${role}" \
+    --password "${USTC_ADMIN_PASS}" || true
+
+  aws cognito-idp admin-confirm-sign-up \
+    --region "${REGION}" \
+    --user-pool-id "${USER_POOL_ID}" \
+    --username "${email}" || true
+
+  response=$(aws cognito-idp admin-initiate-auth \
+    --user-pool-id "${USER_POOL_ID}" \
+    --client-id "${CLIENT_ID}" \
+    --region "${REGION}" \
+    --auth-flow ADMIN_NO_SRP_AUTH \
+    --auth-parameters USERNAME="${email}"',PASSWORD'="${USTC_ADMIN_PASS}")
+  adminToken=$(echo "${response}" | jq -r ".AuthenticationResult.IdToken")
+}
+
+#createAccount [email] [role] [index] [barNumber] [section] [overrideName(optional)] [employer(optional)] [firstName(*optional)] [middleName(optional)] [lastName(*optional)] [suffix(optional)]
+# *optional - only optional when user is NOT irsPractitioner or privatePractitioner
+createAccount() {
+  email=$1
+  role=$2
+  i=$3
+  barNumber=$4
+  section=$5
+  name=${6:-Test ${role}$3}
+  employer=$7
+  firstName=$8
+  middleName=$9
+  lastName=${10}
+  suffix=${11}
+
+  curl --header "Content-Type: application/json" \
+    --header "Authorization: Bearer ${adminToken}" \
+    --request POST \
+    --data "$(generate_post_data "${email}" "${role}" "${barNumber}" "${section}" "${name}" "${employer}" "${firstName}" "${middleName}" "${lastName}" "${suffix}")" \
+      "https://${restApiId}.execute-api.us-east-1.amazonaws.com/${ENV}/users"
+
+  response=$(aws cognito-idp admin-initiate-auth \
+    --user-pool-id "${USER_POOL_ID}" \
+    --client-id "${CLIENT_ID}" \
+    --region "${REGION}" \
+    --auth-flow ADMIN_NO_SRP_AUTH \
+    --auth-parameters USERNAME="${email}"',PASSWORD="Testing1234$"')
+
+  session=$(echo "${response}" | jq -r ".Session")
+
+  if [ "$session" != "null" ]; then
+    aws cognito-idp admin-respond-to-auth-challenge \
+      --user-pool-id  "${USER_POOL_ID}" \
+      --client-id "${CLIENT_ID}" \
+      --region "${REGION}" \
+      --challenge-name NEW_PASSWORD_REQUIRED \
+      --challenge-responses 'NEW_PASSWORD="Testing1234$",'USERNAME="${email}" \
```

Comment:
Just trying to understand how this works. I see that the `ustcadmin` account creates an `adminToken`, which in turn is used to create the migrator@example.com account. I just don't quite get how this account could not be used to gain access to the system.